benchmarks/api/fabric/workloads/compositeTx/read-write-assets.js

```diff
@@ -0,0 +1,129 @@
+     * @param {BlockchainInterface} sutAdapter The adapter of the underlying SUT.
+     * @param {Object} sutContext The custom context object provided by the SUT adapter.
+     * @async
+     */
+    async initializeWorkloadModule(workerIndex, totalWorkers, roundIndex, roundArguments, sutAdapter, sutContext) {
+        await super.initializeWorkloadModule(workerIndex, totalWorkers, roundIndex, roundArguments, sutAdapter, sutContext);
+
+        const args = this.roundArguments;
+        this.chaincodeID = args.chaincodeID ? args.chaincodeID : 'fixed-asset';
+        const assetNumber = args.assets ? parseInt(args.assets) : 2
+        this.assetsPerWorker = Math.floor(assetNumber / totalWorkers);
+        if (this.workerIndex == 0){
+            const reminder = assetNumber % totalWorkers;
+            this.assetsPerWorker += reminder;
+        } 
+
+        this.byteSize = args.byteSize;
+        this.readSize = args.read ? parseInt(args.read) : 1;//TODO throw error or utput the facttha it defaulted to 1
+        this.writeSize = args.write.number ? parseInt(args.write.number) : 1;//TODO throw error or utput the facttha it defaulted to 1
+        this.previouslyRead = args.write.previouslyRead in writeOptions ? args.write.previouslyRead : writeOptions.random;
+
+        this.asset = {
+            docType: this.chaincodeID,
+            content: '',
+            creator: 'client' + this.workerIndex,
+            byteSize: this.byteSize
+        };
+
+        const paddingSize = this.byteSize - bytes(JSON.stringify(this.asset));
+        this.asset.content = 'B'.repeat(paddingSize);
+    }
+
+    /**
+     * Assemble TXs for the round.
+     * @return {Promise<TxStatus[]>}
+     */
+    async submitTransaction() {
+        // Create argument array [readKeys, writeKeys, otherArgs(String)]
+
+        const readKeys = [];
+        const writeKeys = [];
+
+        const totalRandomIds = this.readSize + this.writeSize;
+        const ids = helper.retrieveRandomAssetIdsFromRange(totalRandomIds, 0, this.assetsPerWorker);
+
+        for (let i = 0; i < this.readSize; i++) {
+            // pick one of the randomized items and remove it from future consideration
+            const uuid = ids.shift(); 
+            const key = 'client' + this.workerIndex + '_' + this.byteSize + '_' + uuid;
+            readKeys.push(key);
+        }
+
+        for(let i = 0; i < this.writeSize; i++){
+            if(i < readKeys.length && this.previouslyRead == writeOptions.read){ //previosly read keys write
+                writeKeys.push(readKeys[i]);
+            } else if (this.previouslyRead == writeOptions.random){ //random keys write
+                const uuid = Math.floor(Math.random() * this.assetsPerWorker);
+                const key = 'client' + this.workerIndex + '_' + this.byteSize + '_' + uuid;
+                writeKeys.push(key);
+            } else { //not already read keys write
```

Comment:
This doesn't look like it guarantees that the keys that will be written to have not been read (see comment in the helper file)